docs/glossary.md

```diff
@@ -2,3 +2,57 @@
 
```

Comment:
there are two glossary.md files (this one and the one in 'References'). Do we still need this one?